torch/nested/_internal/ops.py

```diff
@@ -897,16 +894,38 @@ def sum_dim_IntList(func, *args, **kwargs):
                     "sum(): not supported along a ragged and non-batch dimension for NestedTensor"
                 )
             # reduction cases: (ragged)
+            # shift reduction dimension of values backward to outer dimension
+            values_transposed_backward = inp._values.permute(
+                inp._ragged_idx - 1,
+                *range(0, inp._ragged_idx - 1),
+                *range(inp._ragged_idx, inp.dim() - 1),
+            )
+
+            padded = torch.ops.aten._jagged_to_padded_dense_forward(
+                values_transposed_backward.reshape(
+                    values_transposed_backward.shape[0], -1
+                ),  # _jagged_to_padded_dense_forward requires values to be a 2D tensor
```

Comment:
... `_jagged_to_padded_dense_forward requires values to be a 2D tensor with the ragged dimension as the 0th dimension`